src/app/routes/applications/new-role/services/role-creation.service.ts

```diff
@@ -21,10 +21,15 @@ export class RoleCreationService {
   async canUseDomain(domain: string): Promise<boolean> {
     try {
       const exists = await this.domainsFacade.checkExistenceOfDomain(domain);
-      if (!exists) {
-        return true;
+      const isOwner = await this.domainsFacade.isOwner(domain);
+      if (exists) {
+        return false;
+      }
```

Comment:
Slightly more efficient to do it this way as we don't need `isOwner` if `exist == true`
```suggestion
      if (exists) {
        return false;
      }
      const isOwner = await this.domainsFacade.isOwner(domain);
```

Reply:
You're right, I'll update this one.